neurox/interpretation/gaussian_probe.py

```diff
@@ -0,0 +1,191 @@
+        S = torch.stack(S).to(self.device)
+        lambda_n = lambda_0 + S
+        self.mu_star = mu_n
+        sigma_star = lambda_n / (v_n + self.train_features.shape[1] + 2).view(
+            self.labels_dim, 1, 1
+        )
+        min_eig = []
+        for sigma in sigma_star:
+            eigs = torch.eig(sigma).eigenvalues[:, 0]
+            min_eig.append(eigs.min())
+        min_eig = torch.tensor(min_eig).to(self.device)
+        sigma_star[min_eig < 0] -= (
+            min_eig.view(sigma_star.shape[0], 1, 1)[min_eig < 0]
+            * torch.eye(sigma_star.shape[1]).to(self.device)
+            * torch.tensor(10).to(self.device)
+        )
+        self.sigma_star = sigma_star
+
+    def _get_distributions(self, selected_features):
+        self.distributions = []
+        for label in range(self.labels_dim):
+            if self.train_categorical[label].item() == 0:
+                self.distributions.append(torch.distributions.normal.Normal(0.0, 0.0))
+            else:
+                self.distributions.append(
+                    mn.MultivariateNormal(
+                        self.mu_star[label, selected_features].double(),
+                        self.sigma_star[label, selected_features][:, selected_features],
+                    )
+                )
+
+    def _compute_probs(self, selected_features, set_name):
+        features = self.feature_sets[set_name]
+        with torch.no_grad():
+            log_probs = []
+            for i in range(self.labels_dim):
+                if self.train_categorical[i] == 0:
+                    log_probs.append(torch.zeros(features.shape[0], dtype=torch.double))
+                else:
+                    log_probs.append(
+                        self.distributions[i].log_prob(features[:, selected_features])
+                    )
+            log_probs = torch.stack(log_probs, dim=1)
+            log_prob_times_cat = log_probs + self.train_categorical.log()
+            self.not_normalized_probs = log_prob_times_cat
+            self.normalizer = log_prob_times_cat.logsumexp(dim=1)
+        self.probs = log_prob_times_cat - self.normalizer.unsqueeze(1)
+
+    def _predict(self, set_name: str):
+        preds = self.probs.argmax(dim=1)
+        labels = self.label_sets[set_name]
+        accuracy = ((preds == labels)).nonzero(as_tuple=False).shape[0] / labels.shape[
+            0
+        ]
+        categorical = self.categorical_sets[set_name]
+        entropy = torch.distributions.Categorical(categorical).entropy()
+        with torch.no_grad():
+            conditional_entropy = -self.probs[
+                list(range(self.probs.shape[0])), labels
+            ].mean()
+        mutual_inf = (entropy - conditional_entropy) / torch.tensor(2.0).log()
+        return accuracy, mutual_inf.item(), (mutual_inf / entropy).item()
+
+
+def train_probe(X, y):
+    return GaussianProbe(X, y)
+
+
+def evaluate_probe(probe, X_test, y_test, selected_neurons):
```

Comment:
Is it possible to "evaluate" a probe on all neurons? If yes we can make `selected_neurons` optional with the default behavior being evaluation on all neurons